xsl/xccdf-resources/openscap.js

```diff
@@ -179,44 +180,27 @@ function sortGroups(groups, key)
     switch(key) {
     case KeysEnum.SEVERITY:
         return ["high", "medium", "low"];
-    case KeysEnum.DISA:
-        return groups.sort(function(a, b){
-            return parseInt(a) - parseInt(b);
-        });
-    case KeysEnum.NIST:
+    case KeysEnum.RESULT:
+	return groups.sort();
+    default:
         return groups.sort(function(a, b){
-            var regex = /(\w\w)-(\d+)(.*)/;
-            var a_parts = regex.exec(a);
-            var b_parts = regex.exec(b);
-            if (a_parts == null)
-                return 1;
-            if (b_parts == null)
-                return -1;
-            var result = a_parts[1].localeCompare(b_parts[1]);
-            if (result != 0) {
-                return result;
-            } else {
-                result = a_parts[2] - b_parts[2];
-                if (result != 0) {
-                    return result;
+            var a_parts = a.split(/[.()-]/);
```

Comment:
From what I found here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions this is a so-called "regular expression literal". It is compiled when the script is loaded, not at runtime, so it will be compiled only once. It think we can leave it as is.